src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -249,14 +255,22 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
                 $responses['default'] = new Response('Unexpected error');
             }
 
-            if ($contextResponses = $operation->getOpenapiContext()['responses'] ?? false) {
+            if ($contextResponses = $operation->getOpenapi()?->getResponses() ?: false) {
+                foreach ($contextResponses as $statusCode => $contextResponse) {
+                    $responses[$statusCode] = $contextResponse;
+                }
+            } elseif ($contextResponses = $operation->getOpenapiContext()['responses'] ?? false) {
+                // TODO Remove this "elseif" in 4.0
                 foreach ($contextResponses as $statusCode => $contextResponse) {
                     $responses[$statusCode] = new Response($contextResponse['description'] ?? '', isset($contextResponse['content']) ? new \ArrayObject($contextResponse['content']) : null, isset($contextResponse['headers']) ? new \ArrayObject($contextResponse['headers']) : null, isset($contextResponse['links']) ? new \ArrayObject($contextResponse['links']) : null);
                 }
             }
 
             $requestBody = null;
-            if ($contextRequestBody = $operation->getOpenapiContext()['requestBody'] ?? false) {
+            if ($contextRequestBody = $operation->getOpenapi()?->getRequestBody() ?: false) {
```

Comment:
what if getOpenapi is `true` will that work? `$a = true; $a?->test` shows `PHP Warning:  Attempt to read property "test" on bool in php shell code on line 1`.

Reply:
IMHO we should not allow `true` value as it doesn't make any sense. WDYT about trowing an error on the setter (operation, ApiResource) if the value is `true`, or automatically convert it to null?